appsec/appsec.go

```diff
@@ -0,0 +1,32 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+// Package appsec provides application security features in the form of SDK
+// functions that can be manually called to monitor specific code paths and data.
+// Application Security is currently transparently integrated into the APM tracer
+// and cannot be used nor started alone at the moment.
+// You can read more on how to enable and start Application Security for Go at
+// https://docs.datadoghq.com/security_platform/application_security/getting_started/go
+package appsec
+
+import (
+	"golang.org/x/net/context"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/dyngo/instrumentation/httpsec"
+)
+
+// MonitorParsedHTTPBody runs the security monitoring rules on the given *parsed*
+// HTTP request body. The given context must be the HTTP request context as returned
```

Comment:
If the context must be an HTTP request then why not just pass an HTTP request instead? This is quite error prone as it is now.

Lastly, what do you mean by a "parsed" HTTP request body? Can't we just avoid the `interface{}` type?

Reply:
Thanks for looking into this.
> If the context must be an HTTP request then why not just pass an HTTP request instead?

I think I get your point, ensuring what we ask for by asking the user to pass the request directly rather than a context that could potentially be unrelated to the http request.
While this would work I'm not sure this is such a good way to enforce it, because:
- The request could be ill formed as well since it's provided by the user, meaning this is still error prone. The error vector is shifted but still there and it remains the responsibility of the implementation to handle errors properly.
- This becomes less future proof since some frameworks implement their own Request type (example: `contrib/gofiber/fiber.v2` that relies on `fasthttp`).

I'd be of the opinion to keep the prototype as is given that it allows more flexibility for future frameworks handling for AppSec, plus the fact that the user error possibility it introduces would still be there with a different prototype, only different. WDYT?


> Lastly, what do you mean by a "parsed" HTTP request body?

We indeed refer to the unmarshalled content of the request's body. We use `reflect` further down the stack to analyse the types and pass them to the WAF.